### PR TITLE
oauth2-server: remove primitive obsession of access token

### PR DESCRIPTION
**Pull request by mgenov**
Access token is no longer primitive which allows removing of the TTL property as it's calculated by the BearerToken.

### CODE REVIEW

File: oauth2-example-app/src/main/java/com/clouway/oauth2/exampleapp/storage/InMemoryTokens.java [hunk 1]
```diff
@@ -23,12 +23,10 @@
   private final Map<String, BearerToken> tokens = Maps.newHashMap();
   private final Map<String, String> refreshTokenToAccessToken = Maps.newHashMap();
   private final TokenGenerator tokenGenerator;
-  private Duration timeToLive;
 
   @Inject
   public InMemoryTokens(TokenGenerator tokenGenerator, Duration timeToLive) {
```
Comment (qnislavnachev) [inline oauth2-example-app/src/main/java/com/clouway/oauth2/exampleapp/storage/InMemoryTokens.java]: Duration parameter is not used

**Review by qnislavnachev — CHANGES_REQUESTED** [review]

File: oauth2-example-app/src/main/java/com/clouway/oauth2/exampleapp/storage/InMemoryTokens.java (hunk 1 above)
Comment (mgenov) [reply oauth2-example-app/src/main/java/com/clouway/oauth2/exampleapp/storage/InMemoryTokens.java]: Fixed.

File: oauth2-server/src/main/java/com/clouway/oauth2/DateTime.java [hunk 2]
```diff
@@ -1,11 +1,12 @@
 package com.clouway.oauth2;
 
+import java.io.Serializable;
 import java.util.Date;
 
 /**
  * @author Miroslav Genov (miroslav.genov@clouway.com)
  */
-public final class DateTime {
+public final class DateTime implements Serializable {
```
Comment (qnislavnachev) [inline oauth2-server/src/main/java/com/clouway/oauth2/DateTime.java:9]: Why this class implements Serializable ?

Comment (mgenov) [reply oauth2-server/src/main/java/com/clouway/oauth2/DateTime.java:9]: As it acts as a primitive (wrapper of date) someone would be able to use it for serialization (memcached, datastore and etc).

**Comment by qnislavnachev** [comment]
LGTM

**Pull request merged**